beps/0012-metrics-service/README.md

```diff
@@ -0,0 +1,255 @@
+  },
+});
+```
+
+It also provides a method for creating a plugin-scoped `MetricsService` for each registered plugin.
+
+```ts
+interface RootMetricsService extends MetricsService {
+  forPlugin(pluginId: string): MetricsService;
+}
+
+class DefaultRootMetricsService {
+  // ...
+
+  static fromConfig(config: Config): RootMetricsService {
+    return new DefaultRootMetricsService(config);
+  }
+
+  forPlugin(pluginId: string): MetricsService {
+    return new PluginMetricsService({
+      pluginId,
+      serviceName: this.serviceName,
+      serviceVersion: this.serviceVersion,
+    });
+  }
+}
+```
+
+Each plugin-scoped `MetricsService` acquires a meter when it is created by calling `metrics.getMeter` from the `@opentelemetry/api` package. The internals of the service ensure the metric name is namespaced to match the naming conventions outlined in the [naming conventions](#naming-conventions) section.
+
+```ts
+class PluginMetricsService implements MetricsService {
+  // ...
+  constructor({
+    pluginId,
+    serviceName,
+    serviceVersion,
+  }: PluginMetricsServiceOptions) {
+    this.pluginId = pluginId;
+    this.serviceName = serviceName;
+    this.meter = metrics.getMeter(serviceName, serviceVersion);
+  }
+
+  // Ensures consistency across all metrics
+  private prefixMetricName(name: string): string {
+    return `${this.serviceName}.plugin.${this.pluginId}.${name}`;
+  }
+}
+```
+
+### Naming Conventions
+
+All Backstage metrics MUST follow this hierarchical pattern:
+
+`backstage.{scope}.{scope_name}.{category}.{metric_name}`
+
+**Where:**
+
+- `backstage` is the root namespace for all Backstage metrics
+- `{scope}` is the system scope (`pluginId`, `core` for core services)
+- `{scope_name}` is the name of the scope (e.g. `my-plugin`, `catalog`, `scaffolder`, etc.)
+- `{category}` is the logical grouping within the `{scope}`
+- `{metric_name}` is the name of the metric
+
+The metric name will be the name as provided by the plugin author in the form `{category}.{name}`.
+
+```ts
+const rootMetrics = DefaultRootMetricsService.fromConfig(config);
+const pluginMetricsService = rootMetrics.forPlugin('my-plugin');
+
+const metric = pluginMetricsService.createCounter('my_category.my_metric');
+
+metric.add(1);
+
+// ...
+// metric is now available as `backstage.plugin.my-plugin.my_category.my_metric`
+```
+
+#### Scope
+
+The `scope` represents where it belongs in the Backstage ecosystem.
+
+- `pluginId` - A plugin-specific metric (e.g. `backstage.plugin.catalog.entities.count`)
+- `core` - A core service metric (e.g. `backstage.core.database.connections.active`)
+
+##### Plugin-Scoped Metrics
+
+Pattern: `backstage.plugin.{pluginId}.{category}.{metric_name}`
+
+```yaml
+# Examples
+backstage.plugin.catalog.entities.processed.total
+backstage.plugin.scaffolder.tasks.completed.total
+backstage.plugin.techdocs.builds.active
+backstage.plugin.auth.sessions.active.total # todo: technically a core service and a backend plugin
```

Comment:
OK, think I'm following now. But you say `{scope}` is `pluginId` or `core` but here you have the word "plugin"

Reply:
I've fixed this in my latest commit.